backup: TestBackupCompaction failed

backup.TestBackupCompaction [failed](https://mesolite.cluster.engflow.com/invocations/default/8e3aed79-9f83-43c0-a204-ec053696f1d9?testReportRun=1&testReportShard=47&testReportAttempt=1#targets-Ly9wa2cvYmFja3VwOmJhY2t1cF90ZXN0) on master @ [c285f81b65d9cd5cd9659ae6c8546ed8b853e0e8](https://github.com/cockroachdb/cockroach/commits/c285f81b65d9cd5cd9659ae6c8546ed8b853e0e8):


```
=== RUN   TestBackupCompaction
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestBackupCompaction2100315497
    test_log_scope.go:76: use -show-logs to present logs inline
    testutils.go:307: no Invalid Descriptors
    compaction_test.go:411: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupCompaction2100315497
--- FAIL: TestBackupCompaction (191.79s)
=== RUN   TestBackupCompaction/encrypted_backups
    compaction_test.go:301: condition failed to evaluate within 2m0s: expected job status paused, but got succeeded
--- FAIL: TestBackupCompaction/encrypted_backups (124.36s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=47</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupCompaction.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-49234

backup.TestBackupCompaction [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19365035?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19365035?buildTab=artifacts#/) on master @ [c285f81b65d9cd5cd9659ae6c8546ed8b853e0e8](https://github.com/cockroachdb/cockroach/commits/c285f81b65d9cd5cd9659ae6c8546ed8b853e0e8):


```
=== RUN   TestBackupCompaction
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/6d8f0d953bcf81a6c6476724cd8eb3b1/logTestBackupCompaction4283087489
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    testutils.go:307: no Invalid Descriptors
    compaction_test.go:411: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/6d8f0d953bcf81a6c6476724cd8eb3b1/logTestBackupCompaction4283087489
--- FAIL: TestBackupCompaction (156.06s)
=== RUN   TestBackupCompaction/encrypted_backups
    compaction_test.go:301: condition failed to evaluate within 2m0s: expected job status paused, but got succeeded
--- FAIL: TestBackupCompaction/encrypted_backups (122.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupCompaction.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

backup.TestBackupCompaction [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19366435?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19366435?buildTab=artifacts#/) on master @ [06a3a909658282e8f1090733623e280c17636506](https://github.com/cockroachdb/cockroach/commits/06a3a909658282e8f1090733623e280c17636506):


```
=== RUN   TestBackupCompaction
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/6d8f0d953bcf81a6c6476724cd8eb3b1/logTestBackupCompaction1699885483
    test_log_scope.go:76: use -show-logs to present logs inline
    testutils.go:307: no Invalid Descriptors
    compaction_test.go:411: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/6d8f0d953bcf81a6c6476724cd8eb3b1/logTestBackupCompaction1699885483
--- FAIL: TestBackupCompaction (148.61s)
=== RUN   TestBackupCompaction/encrypted_backups
    compaction_test.go:301: condition failed to evaluate within 2m0s: expected job status paused, but got succeeded
--- FAIL: TestBackupCompaction/encrypted_backups (121.74s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupCompaction.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

backup.TestBackupCompaction [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19367041?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19367041?buildTab=artifacts#/) on master @ [ac8745da19fedfb061eb2128bddbc6876f3b13af](https://github.com/cockroachdb/cockroach/commits/ac8745da19fedfb061eb2128bddbc6876f3b13af):


```
=== RUN   TestBackupCompaction
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/6d8f0d953bcf81a6c6476724cd8eb3b1/logTestBackupCompaction3839388204
    test_log_scope.go:76: use -show-logs to present logs inline
    testutils.go:307: no Invalid Descriptors
    compaction_test.go:411: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/6d8f0d953bcf81a6c6476724cd8eb3b1/logTestBackupCompaction3839388204
--- FAIL: TestBackupCompaction (154.39s)
=== RUN   TestBackupCompaction/encrypted_backups
    compaction_test.go:301: condition failed to evaluate within 2m0s: expected job status paused, but got succeeded
--- FAIL: TestBackupCompaction/encrypted_backups (122.40s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupCompaction.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/backup/compaction_test.go
+++ b/pkg/backup/compaction_test.go
@@ -282,7 +282,7 @@ ARRAY['nodelocal://1/backup/%d'], '%s', ''::BYTES, %d::DECIMAL, %d::DECIMAL
 		var jobID jobspb.JobID
 		pause := rand.Intn(2) == 0
 		if pause {
-			db.Exec(t, "SET CLUSTER SETTING jobs.debug.pausepoints = 'backup.after.details_has_checkpoint'")
+			db.Exec(t, "SET CLUSTER SETTING jobs.debug.pausepoints = 'backup_compaction.after.details_has_checkpoint'")
 		}
 		db.QueryRow(
 			t,

